src/CodeGen_Hexagon.cpp

```diff
@@ -1716,7 +1716,6 @@ void CodeGen_Hexagon::visit(const Call *op) {
             internal_assert(op->args.size() == 2);
             string instr = op->is_intrinsic(Call::shift_left) ? "halide.hexagon.shl" : "halide.hexagon.shr";
             Expr b = maybe_scalar(op->args[1]);
-            internal_assert(b.type().is_uint());
```

Comment:
Do we need to assert ```b.type().is_uint() || b.type().is_int()``` now?

Reply:
I think doing Dillon's suggestion fixed the issue here.